packages/babel-core/src/config/files/configuration.js

```diff
@@ -24,6 +24,21 @@ const BABELRC_FILENAME = ".babelrc";
 const BABELRC_JS_FILENAME = ".babelrc.js";
 const BABELIGNORE_FILENAME = ".babelignore";
 
+export function findConfigUpwards(rootDir: string): string | null {
+  let dirname = rootDir;
+  while (true) {
+    if (fs.existsSync(path.join(dirname, BABEL_CONFIG_JS_FILENAME))) {
+      return dirname;
+    }
+
+    const nextDir = path.dirname(dirname);
+    if (dirname === nextDir) break;
+    dirname = nextDir;
```

Comment:
Maybe we could set a max limit, we can be almost sure to not find any config after 30 iterations. That will avoid tranversing the whole system in some special setups.

Reply:
We do this type of look in several places during config resolution. I'm not sure it's worth it to over-optimize up front, especially since you have to opt into the `upward-optional` mode for it to have been wasted time.

Reply:
Ok that make sense, i'm just worried about some undebuggable issues that it could cause.